### DISCUSSION TITLE
KeyFrames migration to V9

### DISCUSSION THREAD

**Discussion by raghavbk** [pmndrs/react-spring#1689] [Support] [answered]
Hi We were using KeyFrames.Spring to animate our charts.  No we are stuck in migrating to V9 as KeyFrames are removed in V9, can someone guide how can we achieve following animation in V9

Working animation V8
https://codesandbox.io/s/react-chart-bh1j2

**Comment by joshuaellis** [comment, marked as answer]
Take a look at https://react-spring.io/hooks/use-spring#async-chainsscripts it's very powerful. And essentially gives you keyframes (i guess)

**Reply by raghavbk** [reply]
@joshuaellis thank you able to solve with `<Spring>` HOC
